cdap-docs/admin-manual/source/operations/logging.rst

```diff
@@ -1,77 +1,573 @@
+    certain messages.
+
+
+User Application Program Logs
+=============================
+
+Emitting Log Messages from a Program
+------------------------------------
 .. highlight:: java
 
-CDAP supports logging through standard `SLF4J (Simple Logging Facade for Java)
-<http://www.slf4j.org/manual.html>`__ APIs. For instance, in a flowlet you can write::
+CDAP supports application logging through the standard `SLF4J (Simple Logging Facade for Java)
+<http://www.slf4j.org/manual.html>`__ APIs.
+
+For instance, in a flowlet you can write::
 
   private static Logger LOG = LoggerFactory.getLogger(WordCounter.class);
   ...
   @ProcessInput
   public void process(String line) {
-    LOG.info("{}: Received line {}", this.getContext().getTransactionAwareName(), line);
+    LOG.debug("{}: Received line {}", this.getContext().getTransactionAwareName(), line);
     ... // processing
-    LOG.info("{}: Emitting count {}", this.getContext().getTransactionAwareName(), wordCount);
+    LOG.debug("{}: Emitting count {}", this.getContext().getTransactionAwareName(), wordCount);
     output.emit(wordCount);
   }
 
-The log messages emitted by your application code can be viewed in different ways.
+This will emit "debug" level messages from a flowlet when it is processing an event.
+
+Retrieving Log Messages from a Program
+--------------------------------------
+The log messages emitted by your application code can be retrieved by:
+
+- Using the :ref:`restful-api`: the :ref:`Logging HTTP RESTful API
+  <http-restful-api-logging>` details the available contexts that can be called to
+  retrieve different messages.
+- Log messages of a program can be viewed in the :ref:`CDAP UI <cdap-ui>`. In the
+  *Overview* page, select the application and program that you are interested in, and click
+  its *Logs* icon (circled in red here):
+
+  .. figure:: /_images/logging/logging-cdap-ui-purchase.png
+      :figwidth: 100%
+      :width: 800px
+      :align: center
+      :class: bordered-image
+
+      **CDAP UI:** *Log* icon, enabled if there are logs available for viewing 
+
+  The logs will be displayed in the log viewer:
+
+  .. figure:: /_images/logging/logging-cdap-ui-purchase-flow-log.png
+      :figwidth: 100%
+      :width: 800px
+      :align: center
+      :class: bordered-image
+
+      **CDAP UI:** *Log Viewer*, showing *PurchaseFlow* log events, *INFO* level
+
+Program Log File Locations
+--------------------------
+Program logs are stored in locations specified by properties in the
+:ref:`cdap-site.xml <appendix-cdap-default-logging>` file depending on the mode 
+of CDAP (Standalone or Distributed):
+
+- For **Standalone CDAP:** the property ``log.collection.root`` (default
+  ``${local.data.dir}/logs``) is the root location for collecting program logs when in
+  Standalone CDAP.
+
+- For **Distributed CDAP:** the property ``log.base.dir`` (default ``/logs/avro``) is the
```

Comment:
we don't use `log.base.dir` in distributed anymore. its `<cdap-dir>/logs` cdap-dir is based on cdap/hdfs namespace config.